### PR TITLE
Clearer doc

### PR DESCRIPTION
**Pull request by bratseth**
@vekterli please review

I'm getting a bit tired of debugging peoples client implementations. This is to try to make it simpler to get it right.

### CODE REVIEW

File: documentation/vespa-http-client.html [hunk 1]
```diff
@@ -63,134 +62,73 @@ <h2 id="using-from-the-command-line">Using from the command line</h2>
...
+
 <p>
-In this API, call <code>stream(document)</code> on document operations.
-The API will block while/if the queues are full.
-When a document has been processed, a callback is asynchronously made.
-This callback will always happen, even when errors occur.
+This section explains the four steps needed to use the client API from your own Java code. For maximum throughput
+this is an async API. In some cases, especially when feeding data from systems which do not support ack'ing of individual
+operations such as e.g Kafka and DynamoDb, it is possible to use the SyncFeedClient wrapper instead, which blocks and
+returns a single result for a batch of operations. The throughput reduction from using this recreases with
```
Comment (jobergum) [inline documentation/vespa-http-client.html]: decreases

Comment (bratseth) [reply documentation/vespa-http-client.html]: Fixed - thanks!

File: documentation/vespa-http-client.html [hunk 2]
```diff
@@ -63,134 +62,73 @@ <h2 id="using-from-the-command-line">Using from the command line</h2>
...
-</p><p>
-The <code>Result</code> has information on its document ID,
-the Endpoint it was fed to (recall that operations are sent to <code>Endpoints</code>
-based on hashing of document ID within one Cluster),
-whether it was successful or not, whether an error was transient or permanent,
-and any exception that was thrown (possibly containing an error message from the Vespa cluster).
-</p><p>
-The order of the callbacks is non-deterministic -
-the only guarantee is that operations are sequenced per document ID.
+Exactly one <code>oncompletion</code> callback will always be received from the client for each operation
```
Comment (vekterli) [inline documentation/vespa-http-client.html]: `s/oncompletion/onCompletion/`

File: documentation/vespa-http-client.html [hunk 3]
```diff
@@ -63,134 +62,73 @@ <h2 id="using-from-the-command-line">Using from the command line</h2>
...
+passed in a stream call, whether or not it was successful.
+Implementing the onCompletion method of the callback is how you can know
+whether or not your feed operations were successfully persisted in Vespa or not. The usual way to implement
```
Comment (vekterli) [inline documentation/vespa-http-client.html]: Consider removing one of the instances of "or not"

**Comment by bratseth** [comment]
All done; thanks

**Pull request merged**